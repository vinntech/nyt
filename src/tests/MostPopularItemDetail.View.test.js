import React from 'react';
import renderer from 'react-test-renderer';
import MostPopularItemDetailView from '../components/article-detail/MostPopularItemDetail.View';

describe("Snapshot", () => {

  test("Snapshot matches to atricles controller", () => {
    const props = {
        selectedArticle:{

            title:"Title",
            byline:"Line",
            avtar:"/avtar.jpg",
            published_date:"12-12-2019",
            mediaImage:"/test.jpg",
            abstract:"more details"
        }
    };
    const appComponent = renderer.create(<MostPopularItemDetailView {...props} />);
    const appComponentJson = appComponent.toJSON();
    expect(appComponentJson).toMatchSnapshot();
  });

});

// describe("Required functions and properties exists" , () => {
//     test("Function:getAvtrarThumbnailUrl exists", () => {
//         expact(MostPopularItemDetail).toHaveProperty("state");
//     });

// });

