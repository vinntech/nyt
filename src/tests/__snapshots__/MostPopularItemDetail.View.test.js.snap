// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Snapshot matches to atricles controller 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiCard-root"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        className="MuiAvatar-root"
        yalt="Remy Sharp"
      >
        <img
          className="MuiAvatar-img"
          src="/avtar.jpg"
        />
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiTypography-body2 MuiTypography-displayBlock MuiCardHeader-title"
      >
        Title
      </span>
      <span
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock MuiCardHeader-subheader"
      >
        Line
      </span>
    </div>
    <div
      className="MuiCardHeader-action"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        12-12-2019
      </p>
    </div>
  </div>
  <img
    className="MuiCardMedia-root MuiCardMedia-media"
    src="/test.jpg"
    title="Title"
  />
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      more details
    </p>
  </div>
</div>
`;
